docs/settings_ref.rst

```diff
@@ -497,3 +497,110 @@ PROXIES
 
 An optional proxy for all communication with the server. Example: ``{'http': '10.0.0.1', 'https': '10.0.0.2'}``
 See the `requests documentation <https://requests.readthedocs.io/en/v3.0.0/api/#requests.Session.proxies>`__ for more information.
+
+TOKEN_REFRESH_THRESHOLD
+---------------------------
+* **Default**: ``300`` (5 minutes)
+* **Type**: ``integer``
+* **Unit**: seconds
+
+Used by the ``TokenLifecycleMiddleware`` to determine how long before token expiration to attempt a refresh.
+This setting controls how proactively the middleware will refresh tokens before they expire.
+
+For example, with the default value of 300 seconds (5 minutes), if a token is set to expire in 4 minutes,
+the middleware will attempt to refresh it during the next request. This helps ensure that users don't
+experience disruptions due to token expiration during active sessions.
+
+.. code-block:: python
+
+    # In your Django settings.py
+    # Refresh tokens 10 minutes before they expire
+    AUTH_ADFS = {
+        # other settings
+        "TOKEN_REFRESH_THRESHOLD": 600
+    }
+
+STORE_OBO_TOKEN
```

Comment:
I don't understand the purpose of this setting. Seems like if a person is using `TokenLifecycleMiddleware`, this has to be True.